### PR TITLE
Introduce APL interface

### PR DESCRIPTION
**Pull request by krzysztofwolski**
Introduce a common interface for auth data.

### CODE REVIEW

File: docs/apl.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    return;
+  },
+  set: async (authData: AuthData) => {
+    await client.set(authData.domain, authData.token);
+  },
+  delete: async (domain: string) => {
+    await client.del(domain);
+  },
+  list: async () => {
```
Comment (lkostrowski) [inline docs/apl.md:40]: getAll

File: package.json [hunk 2]
```diff
@@ -53,6 +53,11 @@
       "import": "./const.mjs",
       "require": "./const.js"
     },
+    "./types": {
+      "types": "./types.d.ts",
+      "import": "./types.mjs",
+      "require": "./types.js"
```
Comment (lkostrowski) [inline package.json:59]: Can we rather scope by packages instead by file types? It's not cohesive.

We can have `import {APL, ...} from '@saleor/app-sdk/APL'` and avoid mixing thing around the project.

The same I did in #25

File: src/manifest.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+export interface WebhookManifest {
+  name: string;
+  events: string[];
+  query: string;
+  targetUrl: string;
+  isActive: boolean;
+}
+
+export interface Manifest {
```
Comment (lkostrowski) [inline src/manifest.ts:17]: `AppManifest`? If we scope `ExtensionManifest` and `WebhookManifest`, maybe this also should be very explicit

Comment (lkostrowski) [reply src/manifest.ts:17]: Also - maybe not in this PR - but I'd like to have Manifest strongly validated

File: src/middleware.ts [hunk 4]
```diff
@@ -187,3 +188,19 @@ export const withJWTVerified =
 
     return handler(request);
   };
+
+export const withExistingRegistration =
```
Comment (lkostrowski) [inline src/middleware.ts:192]: 1. Can we work on the naming better? I'm not sure if with is clear
2. Can we have some folder `middleware/` and there 1 file per middleware + test + index that exports them?

File: src/middleware.ts [hunk 5]
```diff
@@ -187,3 +188,19 @@ export const withJWTVerified =
...
+  (apl: APL): Middleware =>
```
Comment (lkostrowski) [inline src/middleware.ts:193]: can we decide that we will use `{apl: APL}` object for named parameters? It's not only better from DX perspective (I know exlicitly what I pass there), but also will protect us from breaking changes and multiple arguments in the future (we can just extend object)

File: src/types/apl.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export interface AuthData {
+  domain: string;
+  token: string;
+}
+
+export interface APL {
```
Comment (lkostrowski) [inline src/types/apl.ts:6]: This also I'd move to `src/APL/apl.ts` and near it, we can have `src/APL/local-env-apl.ts` and other implementations

**Review by lkostrowski — CHANGES_REQUESTED** [review]

File: src/manifest.ts (hunk 3 above)
Comment (krzysztofwolski) [reply src/manifest.ts:17]: Was not intended to be added here. Removing from the scope.

File: src/middleware.ts (hunk 4 above)
Comment (krzysztofwolski) [reply src/middleware.ts]: 1. Changed to a more explicit name, is it better?
2. Okay! Can it be a separate PR imminently after this one? In this one, I've followed the existing structure and would like to pollute APL changes with middleware refactor

Comment (krzysztofwolski) [reply src/middleware.ts]: Issue for middleware [#29](https://github.com/saleor/saleor-app-sdk/issues/29)

File: src/middleware.test.ts [hunk 7]
```diff
@@ -29,4 +32,54 @@ describe("middleware.test.ts", () => {
...
+      const mockRequest = {
+        context: {},
+        headers: {
+          host: "my-saleor-env.saleor.cloud",
+          "x-forwarded-proto": "https",
+          [SALEOR_DOMAIN_HEADER]: "not-registered.example.com",
+        },
+      } as unknown as Request;
+
+      await withRegisteredSaleorDomainHeader({ apl: mockAPL })(mockHandlerFn)(mockRequest);
```
Comment (lkostrowski) [inline src/middleware.test.ts:80]: I think you can actually assert if error response was returned - this is client facing stuff

**Review by lkostrowski — APPROVED** [review]

**Pull request merged**